scripts/utilities.py

```diff
@@ -0,0 +1,81 @@
+#-*- coding: utf-8-unix -*-
+import re
+
+
+def get_version_from_gradle():
+    with open("versions.gradle", "r") as f:
+        for line in f:
+            if "version" in line:
+                match = re.search(r"version\s*=\s*['\"]([^'\"]+)['\"]", line)
+                if match:
+                    return match.group(1)
+    raise ValueError("Version not found in versions.gradle")
+
+
+def replace_issue_numbers(commit_msg):
+    """ Function to replace issue numbers with markdown link
+    such as:
+    #132 -> [#132](https://github.com/casualcore/casual-java/issues/132)
+    """
+    pattern = r'#(\d+)'
+
+    def replacement(match):
+        number = match.group(1)
+        return f'[{match.group(0)}](https://github.com/casualcore/casual-java/issues/{number})'
```

Comment:
This can be retrieved from the event information.
`github.event.repository.issues_url`
The format of the url you receive seems to conform to RFC 6570 URI templating, so `https://github.com/casualcore/casual-java/issues{/number}`, which you should be able to format nicely.

The only slight wrinkle I see is that it needs to be slightly modified as that is the api link, rather than the html link.
We also recieve the api and server urls in the same event payload so can easily fix that too:
Just run a replace on the link with the value from:
`github.event.api_url`
with:
`github.event.server_url`

Reply:
If you want to see the full details.
```
      - name: Update changelog
        env:
          GITHUB_CONTEXT: ${{ toJson(github)}}
        run: |
          echo "Hi there, here is the context:"
          echo "$GITHUB_CONTEXT"
```

Reply:
Looks like a lot of the data is actually already available as envs under `GITHUB_`, i just ran with `env | grep GITHUB | sort` and saw `GITHUB_API_URL` and `GITHUB_SERVER_URL` are already there :)